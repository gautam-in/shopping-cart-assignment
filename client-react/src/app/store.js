import { configureStore } from '@reduxjs/toolkit';

import bannerReducer from '../features/banner/BannerSlice'
import categoryReducer from '../features/categories/CategorySlice'
import productReducer from '../features/productList/ProductListSlice'
import cartReducer from '../features/cart/CartSlice'

export const store = configureStore({
  reducer: {
    banner: bannerReducer,
    category: categoryReducer,
    product: productReducer,
    cart: cartReducer
  },
});

