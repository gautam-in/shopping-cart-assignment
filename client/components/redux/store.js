import { applyMiddleware, createStore } from "redux";
import { composeWithDevTools } from "redux-devtools-extension";
import thunk from "redux-thunk";

import addItemReducer from "./addItem/addItemReducer";
import rootReducer from "./rootReducer";

const store = createStore(
  rootReducer,
  composeWithDevTools(applyMiddleware(thunk))
);
/* const store = createStore(addItemReducer, composeWithDevTools()); */

export default store;
