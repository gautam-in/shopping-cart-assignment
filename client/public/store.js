import React, {
  useReducer,
  createContext,
  useContext,
  useCallback,
} from "react";

const StateContext = createContext();
const DispatchContext = createContext();

export const StoreProvider = ({ children }) => {
  const reducer = useCallback((state, action) => {
    const newState = { ...state };
    const productIndex = state.cart?.items?.findIndex(
      (val) => val.id == action.payload.id
    );
    console.log("state", state, action.payload.id, productIndex);
    switch (action.type) {
      case "ADD_TO_CART":
        if (productIndex < 0) {
          add(newState, action.payload);
        } else {
          increament(newState, productIndex);
        }
        return newState;
      case "INCREAMENT":
        increament(newState, productIndex);
        return newState;
      case "DECREAMENT":
        decreament(newState, productIndex);
        return newState;
      case "REMOVE":
        remove(newState, productIndex);
        return newState;
      case "USER":
        return { ...state, ...action.payload };
      case "TOGGLE_CART":
        return { ...state, ...action.payload };
      case "CLOSE_CART":
        return { ...state, ...action.payload };
      case "DO_CART_EMPTY":
        return { ...state, ...action.payload };
      default:
        throw new Error("unknown Wrror");
    }
  }, []);
  // const UserCache = localStorageLoggedInuser();
  const [cartState, dispatch] = useReducer(reducer, {
    cart: { items: [] },
    user: {},
    cartOpen: false,
  });

  return (
    <DispatchContext.Provider value={dispatch}>
      <StateContext.Provider value={cartState}>
        {children}
      </StateContext.Provider>
    </DispatchContext.Provider>
  );
};

function increament(newState, productIndex) {
  newState.cart.items[productIndex].quantity += 1;
}
function decreament(newState, productIndex) {
  newState.cart.items[productIndex]?.quantity > 1
    ? (newState.cart.items[productIndex].quantity -= 1)
    : remove(newState, productIndex);
}
function remove(newState, productIndex) {
  const filteredItems = newState.cart.items.filter(
    (item) => item.id != newState.cart.items[productIndex].id
  );
  newState.cart.items = [...filteredItems];
}
function add(newState, payload) {
  newState.cart.items.push(payload);
}

export const useStore = () => {
  const cartState = useContext(StateContext);
  const dispatch = useDispatchStore();
  const { cart, user, cartOpen } = cartState;
  const { items } = cart;
  const total = items.reduce((acc, item) => {
    return acc + item.quantity * item.price;
  }, 0);
  const totalquantity = items.reduce((acc, item) => {
    return acc + item.quantity;
  }, 0);
  const setToggleCart = () => {
    dispatch({
      type: "TOGGLE_CART",
      payload: { cartOpen: !cartOpen },
    });
  };
  const setCloseCart = () => {
    dispatch({
      type: "CLOSE_CART",
      payload: { cartOpen: false },
    });
  };
  const doCartEmpty = () => {
    dispatch({
      type: "DO_CART_EMPTY",
      payload: { cart: { items: [] } },
    });
  };
  return {
    items,
    total,
    totalquantity,
    user,
    cartOpen,
    setToggleCart,
    setCloseCart,
    doCartEmpty,
  };
};
export const useDispatchStore = () => useContext(DispatchContext);
