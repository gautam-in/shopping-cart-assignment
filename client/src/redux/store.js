import { configureStore, combineReducers } from "@reduxjs/toolkit";
import authReducer from "./slice/authSlice";

const rootReducer = combineReducers({
  auth: authReducer,
});

const store = configureStore({
  reducer: rootReducer,
});

export default store;
