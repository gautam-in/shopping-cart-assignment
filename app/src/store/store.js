import { compose, createStore, applyMiddleware } from 'redux';
import logger from 'redux-logger'
import { rootReducer } from "./root-reducer";

const middleware = [logger];

const composedEnhancers = compose(applyMiddleware(...middleware))

export const store = createStore(rootReducer,undefined,composedEnhancers)