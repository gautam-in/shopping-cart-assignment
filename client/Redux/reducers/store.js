import { applyMiddleware, combineReducers, createStore } from "redux";
import thunk from "redux-thunk";
import { cartReducer } from "./cartReducer";
import { categoryReducer } from "./categoryReducer";
import { errorReducer } from "./errorReducer";
import { userReducer } from "./userReducer";

const store = createStore(
  combineReducers({
    cart: cartReducer,
    categories: categoryReducer,
    user: userReducer,
    error: errorReducer
  }),
  applyMiddleware(thunk)
);

export default store;
