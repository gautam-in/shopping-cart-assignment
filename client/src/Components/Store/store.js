import {createStore } from 'redux'
import CombineReducers from './CombineReducers'

const reduxStore = createStore(CombineReducers)

export default reduxStore