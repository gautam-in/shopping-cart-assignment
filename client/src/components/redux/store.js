import { createStore,applyMiddleware } from "redux"
import {logger} from "redux-logger"
import reduxThunk from "redux-thunk"
import rootReducer from "./rootReducer" 

const middlewares=[reduxThunk];

if(process.env.NODE_ENV==="development"){
    middlewares.push(logger)
}

const store=createStore(rootReducer,applyMiddleware(...middlewares))

export default store;

