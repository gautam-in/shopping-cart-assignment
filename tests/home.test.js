import Component from "../pages/index";
import { mount } from "enzyme";
import { Provider } from "react-redux";
import store from "../Redux/reducers/store";

 
describe("Home page", () => {
  const wrapper = mount(
    <Provider store={store}>
      <Component />
    </Provider>
  );
  test("Home Component should match Snapshot",() => {
    expect(wrapper).toMatchSnapshot();
  })
})