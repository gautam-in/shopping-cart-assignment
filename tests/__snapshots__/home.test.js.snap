// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page Home Component should match Snapshot 1`] = `
<Provider
  store={
    Object {
      "@@observable": [Function],
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <home>
    <HomeComponent>
      <ErrorComponent
        errorMessage="Unable to load banners."
      >
        <GlobalStyle__Container>
          <main
            className="GlobalStyle__Container-sc-hvi81o-0 fMUedy"
          >
            <ErrorStyle__ErrorStyleMain>
              <main
                className="ErrorStyle__ErrorStyleMain-sc-vd5rkk-0 buviRQ"
              >
                <div
                  className="error"
                >
                  <h5 />
                  <h5>
                    Unable to load banners.
                  </h5>
                </div>
              </main>
            </ErrorStyle__ErrorStyleMain>
          </main>
        </GlobalStyle__Container>
      </ErrorComponent>
      <HomeCategory>
        <ErrorComponent
          errorMessage="Unable to load products categories"
        >
          <GlobalStyle__Container>
            <main
              className="GlobalStyle__Container-sc-hvi81o-0 fMUedy"
            >
              <ErrorStyle__ErrorStyleMain>
                <main
                  className="ErrorStyle__ErrorStyleMain-sc-vd5rkk-0 buviRQ"
                >
                  <div
                    className="error"
                  >
                    <h5 />
                    <h5>
                      Unable to load products categories
                    </h5>
                  </div>
                </main>
              </ErrorStyle__ErrorStyleMain>
            </main>
          </GlobalStyle__Container>
        </ErrorComponent>
      </HomeCategory>
    </HomeComponent>
  </home>
</Provider>
`;
